SpringVoting/src/voting/controllers/MainController.java

```diff
@@ -0,0 +1,105 @@
+package voting.controllers;
+
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import javax.validation.Valid;
+import static org.springframework.web.bind.annotation.RequestMethod.*;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataAccessException;
+import org.springframework.security.authentication.AnonymousAuthenticationToken;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.web.authentication.WebAuthenticationDetails;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.ModelAndView;
+
+import voting.dao.UserDAO;
+import voting.daoManagers.UserManager;
+import voting.model.SignInUser;
+import voting.model.SignUpUser;
+import voting.util.Roles;
+
+@Controller
+public class MainController {
+	@Autowired
+	private UserDAO userDAO;
+	@Autowired
+	private UserManager userManager;
+	@Autowired
+	private AuthenticationManager authManager;
+	
+	
+	@RequestMapping(value="/sign", method  = GET)
+	public ModelAndView sign(@RequestParam(value="error", required = false) String error, HttpSession session){		
+		ModelAndView model = new ModelAndView();
+		SignInUser signInUser = new SignInUser();		
+		if (error != null) {
+			signInUser.setEmail(session.getAttribute("LAST_USERNAME").toString());
+			session.removeAttribute("LAST_USERNAME");
```

Comment:
you repeate the same value twice `LAST_USERNAME` - a good point to move this value to constants